File: works/tmp/10997.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.IO;
using static System.Console;
using System.Text;

namespace ConsoleApp
{
    class Program
    {
        static int N;
        static int rSize;
        static int cSize;

        static void Draw(int r, int c, int cnt, int depth, char[,] stars)
        {
            stars[r, c] = '*';

            for (int i = 0; i < cnt - 1; i++)
            {
                r--;
                stars[r, c] = '*';
            }
            for (int i = 0; i < cnt - 1; i++)
            {
                c++;
                if (c >= cSize) break;
                stars[r, c] = '*';
            }

            if (depth == N - 1) return;

            for (int i = 0; i < cnt + 2 - 1; i++)
            {
                r++;
                stars[r, c] = '*';
            }

            for (int i = 0; i < cnt + 2 - 1; i++)
            {
                c--;
                stars[r, c] = '*';
            }

            Draw(r, c, cnt + 4, depth + 1, stars);
        }

        static void Print(char[,] stars)
        {
            StringBuilder sb = new StringBuilder();

            for (int i = 0; i < rSize; i++)
            {
                if (i == 1)
                    sb.Append('*').Append('\n');
                else
                {
                    for (int j = 0; j < cSize; j++)
                        sb.Append(stars[i, j]);
                    sb.Append('\n');
                }
            }

            Write(sb);
        }

        static void Main(string[] args)
        {
            StreamReader sr = new StreamReader(Console.OpenStandardInput());
            N = int.Parse(sr.ReadLine());
            if (N == 1) { Write('*'); return; }

            rSize = 4 * N - 2 + 1;
            cSize = 4 * N - 4 + 1;
            char[,] stars = new char[rSize, cSize];

            for (int i = 0; i < rSize; i++)
            {
                for (int j = 0; j < cSize; j++)
                {
                    stars[i, j] = ' ';
                }
            }

            Draw(2 * N, 2 * N - 2, 3, 0, stars);
            Print(stars);
        }

    }
}

File: works/tmp/10997.py

def Draw(r, c, cnt, depth, stars, N):
    stars[r][c] = '*'

    for _ in range(cnt-1):
        r -= 1
        stars[r][c] = '*'

    for _ in range(cnt-1):
        c += 1
        if c >= len(stars[r]): break
        stars[r][c] = '*'

    if depth == N-1: return

    for _ in range(cnt+2-1):
        r += 1
        stars[r][c] = '*'

    for _ in range(cnt+2-1):
        c -= 1
        stars[r][c] = '*'

    Draw(r, c, cnt+4, depth+1, stars, N)


def Print(stars):
    for i in range(len(stars)):
        if i == 1:
            print('*', end='')
        else :
            for j in range(len(stars[i])):
                print(stars[i][j], end='')
        print()


N = int(input())
if N == 1: print('*')
else:
    stars = [[' ' for _ in range(4*N-4+1)] for _ in range(4*N-2+1)]
    Draw(2*N, 2*N-2, 3, 0, stars, N)
    Print(stars)